openslides_backend/action/actions/user/create.py

```diff
@@ -47,8 +50,42 @@ class UserCreate(
     )
 
     def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        if not (
+            instance.get("username")
+            or instance.get("first_name")
+            or instance.get("last_name")
+        ):
+            raise ActionException("Need username or first_name or last_name")
+        if not instance.get("username"):
+            instance["username"] = self.generate_username(instance)
         if not instance.get("default_password"):
             instance = self.generate_and_set_password(instance)
         else:
             instance = self.set_password(instance)
         return super().update_instance(instance)
+
+    def generate_username(self, instance: Dict[str, Any]) -> str:
+        count = 0
+
+        while True:
+            possible_space = (
+                " " if instance.get("first_name") and instance.get("last_name") else ""
+            )
+            new_username = (
+                instance.get("first_name", "")
+                + possible_space
+                + instance.get("last_name", "")
+            )
+            if count > 0:
+                new_username += str(count)
```

Comment:
Please also add a space before the number to stay consistent with OS3

Reply:
Done.